components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/NewApiVersionMailNotifier.java

```diff
@@ -0,0 +1,151 @@
+/*
+ *
+ *   Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *   WSO2 Inc. licenses this file to you under the Apache License,
+ *   Version 2.0 (the "License"); you may not use this file except
+ *   in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+package org.wso2.carbon.apimgt.core.impl;
+
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.core.configuration.models.APIMConfigurations;
+import org.wso2.carbon.apimgt.core.configuration.models.MailConfigurations;
+import org.wso2.carbon.apimgt.core.exception.APIManagementException;
+import org.wso2.carbon.apimgt.core.template.dto.NotificationDTO;
+
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Properties;
+import java.util.Set;
+import javax.mail.Authenticator;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.PasswordAuthentication;
+import javax.mail.Session;
+import javax.mail.Transport;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+
+
+/**
+ * Class to send Email notification.
+ */
+public class NewApiVersionMailNotifier extends Notifier {
+    private static final Log log = LogFactory.getLog(NewApiVersionMailNotifier.class);
+    static MailConfigurations mailConfigurations = new APIMConfigurations().
+            getNotificationConfigurations().getMailConfigurations();
+    private static final String SMTP_HOST_NAME = mailConfigurations.getSmtpHostName();
+    private static final String SMTP_AUTH_USER = mailConfigurations.getSmtpAuthUser();
+    private static final String SMTP_AUTH_PWD = mailConfigurations.getSmtpAuthPwd();
+
+
+    @Override
+    public void sendNotifications(NotificationDTO notificationDTO) throws
+            APIManagementException {
+
+        Properties props = new Properties();
+        props.put("mail.smtp.host", SMTP_HOST_NAME);
+        props.put("mail.smtp.auth", "true");
+        props.put("mail.smtp.starttls.enable", "true");
+        props.put("mail.smtp.port", "587");
+        props.put("mail.transport.protocol", "smtp");
+        //get Notifier email List
+        Set<String> sendTo = getEmailNotifierList(notificationDTO);
+
+        if (!sendTo.isEmpty()) {
```

Comment:
This can be refactored like this and reduce a branch.
```

 if (sendTo.isEmpty()) {
log.debug("Email Notifier Set is Empty");
return;
}

for (String mail : sendTo) {

```